install.ps1

```diff
@@ -0,0 +1,601 @@
+        # TODO: Select-String
+        $expectedAirgapChecksum = Find-Checksum -ChecksumFilePath $DestinationPath -Pattern "rke2-images.$suffix.tar.zst"
+    }
+    elseif (Test-Path -Path "$Path/rke2-images.$suffix.tar.gz" -PathType Leaf) {     
+        # TODO: Select-String   
+        $expectedAirgapChecksum = Find-Checksum -ChecksumFilePath $DestinationPath -Pattern "rke2-images.$suffix.tar.gz"
+    }
+
+    return @{ ExpectedChecksum = $expectedChecksum; ExpectedAirgapChecksum = $expectedAirgapChecksum }
+}
+
+# stage_local_tarball stages the local tarball.
+function Copy-LocalTarball() {
+    [CmdletBinding()]
+    param (
+        [Parameter()]
+        [String]
+        $Path,
+        [Parameter()]
+        [String]
+        $DestinationPath
+    )       
+    $archInfo = Get-ArchitectureInfo
+    $suffix = $archInfo.Suffix  
+    Write-InfoLog "staging tarball from $ArtifPathactPath/rke2.$suffix.tar.gz"
+    Copy-Item -Path "$Path/rke2.$suffix.tar.gz" -Destination $DestinationPath -Force
+}
+
+# stage_local_airgap_tarball stages the local checksum hash for validation.
+function Copy-LocalAirgapTarball() {
+    [CmdletBinding()]
+    param (
+        [Parameter()]
+        [String]
+        $Path,
+        [Parameter()]
+        [String]
+        $DestinationPath
+    )   
+
+    $archInfo = Get-ArchitectureInfo
+    $suffix = $archInfo.Suffix  
+    if (!(Test-Path -Path "$Path/rke2-images.$suffix.tar.zst" -PathType Leaf)) {
+        Write-InfoLog "staging zst airgap image tarball from $Path/rke2-images.$suffix.tar.zst"
+        Copy-Item -Path "$Path/rke2-images.$suffix.tar.zst" -Destination $DestinationPath -Force
+        return "zst"
+    }
+    elseif (!(Test-Path -Path "$Path/rke2-images.$suffix.tar.gz" -PathType Leaf)) {
+        Write-InfoLog "staging gzip airgap image tarball from $Path/rke2-images.$suffix.tar.gz"
+        Copy-Item -Path "$Path/rke2-images.$suffix.tar.gz" -Destination $DestinationPath -Force
+        return "gz"
+    }
+}
+
+# verify_tarball verifies the downloaded installer checksum.
+function Test-TarballChecksum() {
+    [CmdletBinding()]
+    param (
+        [Parameter()]
+        [String]
+        $Tarball,
+        [Parameter()]
+        [String]
+        $ExpectedChecksum
+    )
+    Write-InfoLog "verifying tarball"
+    $actualChecksum = (Get-FileHash -Path $Tarball -Algorithm SHA256).Hash.ToLower()
+    if ($ExpectedChecksum -ne $actualChecksum) {
+        Write-FatalLog "download sha256 does not match $ExpectedChecksum, got $actualChecksum"
+    }
+}
+
+# unpack_tarball extracts the tarball, correcting paths as necessary
+function Expand-Tarball() {
+    [CmdletBinding()]
+    param (
+        [Parameter()]
+        [string]
+        $InstallPath,
+        [Parameter()]
+        [String]
+        $Tarball
+    )
+    Write-InfoLog "unpacking tarball file to $InstallPath"
+    New-Item -Path $InstallPath -Type Directory -Force
+    tar xzf "$Tarball" -C "$InstallPath"
```

Comment:
No cmdlet for extracting tarballs, huh

Reply:
No there isn't a commandlet for extracting tarballs.